app/controllers/businesses_controller.rb

```diff
@@ -0,0 +1,41 @@
+class BusinessesController < ApplicationController
+  before_action :set_business, only: [ :show ]
+  before_action :authenticate_user!, only: [ :new, :create, :show, :index ]
+
+  def index
+    @businesses = Business.all
+  end
+
+  def show
+  end
+
+  def new
+    @business = Business.new
+  end
+
+  def create
+    @business = Business.new(business_params)
+    @business.user_id = current_user.id
+
+    respond_to do |format|
+      if @business.save
+        format.html { redirect_to @business, notice: 'Business was successfully created.' }
+        format.json { render :show, status: :created, location: @business }
+      else
+        flash[:error] = 'Error creating business'
```

Comment:
We don't usually add a flash error here, but I thought it would be helpful for a user to know that there was a problem creating a business